virtual_ecosystem/models/soil/pools.py

```diff
@@ -845,9 +855,11 @@ def calculate_nutrient_leaching(
         lmwc=labile_carbon_leaching,
         don=don_leaching,
         dop=dop_leaching,
-        ammonium=ammonium_leaching,
-        nitrate=nitrate_leaching,
-        labile_P=labile_phosphorus_leaching,
+        ammonium=np.where(ammonium_leaching >= 0.0, ammonium_leaching, 0.0),
```

Comment:
Like the question above, I wonder if instead of fixing the FALSE condition to zero, it could be the last remaining amount of ammonium instead. Without fully understanding the code, my gut feeling says instead of zeroing the rates, we would be zeroing the nutrient amount instead.

Reply:
Yeah this is the same as the above, where 0.0 is used to handle the case where ammonium concentration is negative. This is then needed to stop negative leaching from happening, which would be nonsensical (and break the nutrient balance)